src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py

```diff
@@ -248,16 +248,16 @@ def _create_solver_object(
     """Create the solver object that is used."""
     _check_step_tol(step, tol)
     _check_method(method)
-    if step is not None:
-        precond_step = step
+    if precond_step is None:
+        precond_step = step or 1.0
     precond_step = precond_step / float(nsteps)
     _prior = _string2prior(ivp, which_prior, precond_step, **kwargs)
     if tol is not None:
-        stprl = _step2steprule_adap(tol, _prior)
+        stprl = steprule.AdaptiveSteps(tol, _prior.ordint + 1, **kwargs)
         if firststep is None:
             firststep = ivp.tmax - ivp.t0
     else:
-        stprl = _step2steprule_const(step)
+        stprl = steprule.ConstantSteps(step)
         firststep = step
```

Comment:
I remember you saying that you don't see an easy way of extracting this into a function. Why again was that the case? Because there would be too many inputs for this function?

Reply:
The function would take `step`, `_prior` (or at least `localconvrate = _prior.ordint + 1`), `tol`, `firststep`, and something related to `ivp.tmax` and `ivp.tmin`, and it would output `steprule, firststep`.

I have the impression that that's a lot of function arguments, and it grows with any new steprule. I think it would be worth it to improve that aspect, but it seems like more than just moving a few lines around, so I thought I't leave it at the old behavior for now.

If you have a clear idea on how to improve that code let me know! :)

Reply:
Okay that is clearly something very subjective and I have no problems with the current solution. I, personally wouldn't mind a function ```steprule, firststep = _step2steprule(step, tol, firststep, _prior, ivp)``` but I understand why you find that odd :)